source/Calamari/Kubernetes/Commands/Executors/GatherAndApplyRawYamlExecutor.cs

```diff
@@ -151,9 +157,16 @@ private IEnumerable<Resource> ApplyBatchAndReturnResources(GlobDirectory globDir
             foreach (var file in files)
             {
                 log.Verbose($"Matched file: {fileSystem.GetRelativePath(directoryWithTrailingSlash, file)}");
+                log.Verbose(fileSystem.ReadFile(file));
             }
 
-            var result = kubectl.ExecuteCommandAndReturnOutput("apply", "-f", $@"""{globDirectory.Directory}""", "--recursive", "-o", "json");
+            log.Info($"SCME ApplyBatchAndReturnResources: {kubectl.GetHashCode()}");
+            log.Info($"SCME ApplyBatchAndReturnResources EnvVars: '{JsonConvert.SerializeObject(kubectl.environmentVars)}'");
+            log.Info($"SCME ApplyBatchAndReturnResources workingDirectory: '{kubectl.workingDirectory}'");
+            if (!TryExecuteCommandWithVerboseLoggingOnly("get", "namespace", @namespace))
+                TryExecuteCommandWithVerboseLoggingOnly("create", "namespace", @namespace);
+
+            var result = kubectl.ExecuteCommandAndReturnOutput("apply", "-f", $@"""{globDirectory.Directory}""", "--recursive", "-o", "json", "-n", @namespace);
```

Comment:
I tried adding `-n <namespace>` but it won't work if the namespace is defined in the users yaml already (and they don't match).